pkg/querier/queryrange/query_range.go

```diff
@@ -222,6 +222,16 @@ func (prometheusCodec) DecodeRequest(_ context.Context, r *http.Request) (Reques
 	result.Query = r.FormValue("query")
 	result.Path = r.URL.Path
 
+	// Include the headers from http request in prometheusRequest.
+	// We don't include the following header in the codec -
+	// accept-encoding - defaultroundtripper uses the response_compression_enabled setting to inflate/deflate responses
+	for h, hv := range r.Header {
+		if strings.ToLower(h) == "accept-encoding" {
```

Comment:
nit: `strings.EqualFold` is more efficient.

Reply:
Learned something new. Thanks.